filebeat/prospector/log/harvester.go

```diff
@@ -326,7 +344,7 @@ func (h *Harvester) SendStateUpdate() {
 
 	d := util.NewData()
 	d.SetState(h.state)
-	h.forwarder.Outlet.OnEvent(d)
+	h.publishState(d)
```

Comment:
It's important that a harvester only gets closed and releases the file handler if the state is persisted to disk, means this event is acked.